api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -78,6 +78,7 @@ message ExtAuthz {
   // 3. At least one ``authorization response header`` is added to the client request, or is used for
   // altering another client request header.
   //
+  // 4. The filter is not being used as an upstream filter.
   bool clear_route_cache = 6;
```

Comment:
Let's config enforce this and add a test. 

I think we can just add a boolean for upstream vs downstream filter in the constructors of DualInfo and then add this to the checks in config.cc

Reply:
Done.